contrib/k8s.io/client-go/kubernetes/kubernetes.go

```diff
@@ -55,46 +55,40 @@ func wrapRoundTripperWithOptions(rt http.RoundTripper, opts ...httptrace.RoundTr
 
 // RequestToResource parses a Kubernetes request and extracts a resource name from it.
 func RequestToResource(method, path string) string {
-	if !strings.HasPrefix(path, prefixAPI) {
+	c := apiRegexp.FindStringSubmatch(path)
+	if c == nil {
```

Comment:
Can we add a `len` check here as well so we don't panic just in case the number of matches is somehow unexpected?